app/actions/manifest_route_update.rb

```diff
@@ -93,7 +93,9 @@ def find_or_create_valid_route(app, manifest_route, user_audit_info)
               manifest_triggered: true
             )
           elsif route.space.guid != app.space_guid
-            raise InvalidRoute.new('Routes cannot be mapped to destinations in different spaces')
+            # check if route is shared with space
+            spaces = route.shared_spaces
+            raise InvalidRoute.new('Routes cannot be mapped to destinations in different spaces') if spaces.blank? || spaces.none? { |space| space.values[:id] == app.space.id }
```

Comment:
This logic is duplicated in several places now (e.g. https://github.com/cloudfoundry/cloud_controller_ng/blob/main/app/controllers/v3/routes_controller.rb#L365).

Is there a shared place we could implement the check? Maybe a function on the `Route` itself?